com.unity.cinemachine/Runtime/Behaviours/CinemachineCollider.cs

```diff
@@ -326,16 +328,32 @@ public override float GetMaxDampTime()
                         cameraPos, state.HasLookAt ? state.ReferenceLookAt : cameraPos);
 
                     // Apply damping
-                    if (deltaTime >= 0 && VirtualCamera.PreviousStateIsValid)
+                    float dampTime = m_DampingWhenOccluded;
+                    if (deltaTime >= 0 && VirtualCamera.PreviousStateIsValid && m_DampingWhenOccluded + m_Damping > Epsilon)
                     {
-                        displacement = extra.previousDisplacement + Damper.Damp(
-                            displacement - extra.previousDisplacement, 
-                            displacement.sqrMagnitude > extra.previousDisplacement.sqrMagnitude ? m_DampingWhenOccluded : m_Damping,
-                            deltaTime);
+                        // To ease the transition between damped and undamped regions, we damp the damp time
+                        dampTime = displacement.sqrMagnitude > extra.previousDisplacement.sqrMagnitude ? m_DampingWhenOccluded : m_Damping;
+                        if (displacement.sqrMagnitude < Epsilon)
+                            dampTime = extra.previousDampTime - Damper.Damp(extra.previousDampTime, dampTime, deltaTime);
+
+                        var prevDisplacement = Quaternion.Inverse(dampingBypass) * (extra.previousCameraPosition - state.CorrectedPosition);
+                        displacement = prevDisplacement + Damper.Damp(displacement - prevDisplacement, dampTime, deltaTime);
                     }
 
-                    extra.previousDisplacement = displacement;
                     state.PositionCorrection += displacement;
+
+                    // Adjust the damping bypass to account for the displacement
+                    if (state.HasLookAt && VirtualCamera.PreviousStateIsValid)
+                    {
+                        var dir0 = extra.previousCameraPosition - state.ReferenceLookAt;
+                        var dir1 = state.CorrectedPosition - state.ReferenceLookAt;
+                        if (dir0.sqrMagnitude > Epsilon && dir1.sqrMagnitude > Epsilon)
+                            state.PositionDampingBypass = UnityVectorExtensions.SafeFromToRotation(
```

Comment:
Why is PositionDampingBypass in euler angles? It is always used as a quaternion. Every time it is used and saved there is a quat to euler and euler to quat that could be avoided.

Reply:
It's a good question, but that would be an API change.  Maybe we can fix it for CM3.